src/api/crisp/crisp-api.ts

```diff
@@ -37,47 +39,77 @@ const formatSessionText = (courseName: string, status: PROGRESS_STATUS) => {
 };
 
 export const updateCrispProfileAccess = async (
-  email: string,
+  user: IUser,
   partnerAccess: PartnerAccessEntity,
   totalTherapySessionsRedeemed: number,
   totalTherapySessionsRemaining: number,
+  hasFeatureLiveChat: boolean,
+  hasCourses: boolean,
+  courseUserService: CourseUserService,
 ) => {
-  const crispResponse = await getCrispPeopleData(email);
-  const crispData = crispResponse.data.data.data;
-  const partners = crispData['partners'].split('; ');
+  const crispResponse = await getCrispPeopleData(user.email);
+  const hasCrispProfile = crispResponse['reason'] === 'resolved';
 
-  if (partners.indexOf(partnerAccess.partner.name) === -1) {
-    partners.push(partnerAccess.partner.name);
-  }
+  if (!!hasCrispProfile) {
+    const crispData = crispResponse.data.data.data;
+    const partners = crispData['partners'].split('; ');
 
-  const updatedCrispData = {
-    partners: partners.join('; '),
-    therapy_sessions_remaining:
-      partnerAccess.therapySessionsRemaining + totalTherapySessionsRemaining,
-    therapy_sessions_redeemed: partnerAccess.therapySessionsRedeemed + totalTherapySessionsRedeemed,
-  };
+    if (partners.indexOf(partnerAccess.partner.name) === -1) {
+      partners.push(partnerAccess.partner.name);
+    }
 
-  updateCrispProfile(updatedCrispData, email);
+    const updatedCrispData = {
+      partners: partners.join('; '),
+      therapy_sessions_remaining:
+        partnerAccess.therapySessionsRemaining + totalTherapySessionsRemaining,
+      therapy_sessions_redeemed:
+        partnerAccess.therapySessionsRedeemed + totalTherapySessionsRedeemed,
+    };
+
+    updateCrispProfile(updatedCrispData, user.email);
+  } else if (!!hasCrispProfile && hasFeatureLiveChat === false && !partnerAccess.featureLiveChat) {
```

Comment:
this else if would never be reached because the above `if (!!hasCrispProfile)` would be entered instead